lib/puppet/pops/validation/tasks_checker.rb

```diff
@@ -22,7 +22,17 @@ def check_CapabilityMapping(o)
   end
 
   def check_CollectExpression(o)
-    illegalTasksExpression(o)
+    if in_ApplyExpression?
```

Comment:
Collapse to `if in_ApplyExpression? && o.query.is_a?(Puppet::Pops::Model::VirtualQuery)`?